solarforecastarbiter/reference_forecasts/utils.py

```diff
@@ -350,3 +358,32 @@ def check_persistence_compatibility(observation, forecast, index):
             raise ValueError('index=True not supported for forecasts'
                              ' with run_length >= 1day')
     _check_instant_compatibility(observation, forecast)
+
+
+def _limit_persistence_run_time(data_start, max_run_time, forecast):
+    """Get a the last run time that would result in the
+    forecast producing points as limited by PERS_PT_LIMIT.
+
+    Parameters
+    ----------
+    data_start: pandas.Timestamp
+    max_run_time: pandas.Timestamp
+    forecast: datamodel.Forecast
+
+    Returns
+    -------
+    pandas.Timestamp
+        Either max_run_time or a last run time that would limit the forecasts
+        generated to the the limit provided by the PERISTENCE_PT_LIMIT env var
+        or DEFAULT_PERS_PT_LIMIT.
+    """
+    fx_pt_limit = float(os.getenv('PERSISTENCE_PT_LIMIT', DEFAULT_PERS_PT_LIMIT))
+
+    pts_per_run = forecast.run_length / forecast.interval_length
+
+    max_runs = fx_pt_limit / pts_per_run
+
+    max_run_points = max_runs * forecast.run_length
+
+    run_time_limit = data_start + forecast.lead_time_to_start + max_run_points
+    return min(max_run_time, run_time_limit)
```

Comment:
```suggestion
    max_total_run_length = max_runs * forecast.run_length

    max_run_time_by_points = data_start + forecast.lead_time_to_start + max_total_run_length
    return min(max_run_time, max_run_time_by_points)
```

Reply:
Probably going over the line limit but I think these names are more accurate and make the procedure more clear.